modules/alloy/src/main/ts/ephox/alloy/api/component/GuiFactory.ts

```diff
@@ -12,16 +12,22 @@ import * as Component from './Component';
 import { AlloyComponent } from './ComponentApi';
 import { AlloySpec, PremadeSpec, SimpleOrSketchSpec, SketchSpec } from './SpecTypes';
 
-const buildSubcomponents = (spec: SimpleOrSketchSpec): AlloyComponent[] => {
+const buildSubcomponents = (spec: SimpleOrSketchSpec, obsoleted: Optional<SugarElement<Node>>): AlloyComponent[] => {
   const components = Obj.get(spec, 'components').getOr([ ]);
-  return Arr.map(components, build);
+
+  return obsoleted.fold(
+    () => Arr.map(components, build),
+    (obs) => Arr.map(components, (c, i) => {
+      return buildOrPatch(c, Traverse.child(obs, i));
+    })
+  );
 };
 
-const buildFromSpec = (userSpec: SketchSpec): Result<AlloyComponent, string> => {
+const buildFromSpec = (userSpec: SketchSpec, obsoleted: Optional<SugarElement<Node>>): Result<AlloyComponent, string> => {
   const { events: specEvents, ...spec }: SketchSpec = CustomSpec.make(userSpec);
 
   // Build the subcomponents. A spec hierarchy is built from the bottom up.
```

Comment:
We might want to add some notes to some of these comments explaining briefly what obsoleted is for.

Reply:
Done